Assets/SteamVR_Unity_Toolkit/Scripts/Controls/3D/Utilities/VRTK_ContentHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace VRTK
+{
+    using UnityEngine;
+
+    /// <summary>
+    /// This script will make objects inside of control contents properly interactable by reparenting them. 
+    /// Otherwise they would disappear if e.g. a drawer is closed even if the object was outside the drawer.
+    /// We need to go through collisions instead of hooking to the Ungrabbed event since ungrabbing will most likely happen above the inside causing the object to drop first. 
+    /// This approach is much more stable and allows for many more scenarios.
+    /// Not supported yet: detect if an object is taken from one content manager and dropped into another one.
+    /// </summary>
+    public class VRTK_ContentHandler : MonoBehaviour
+    {
+        public VRTK_Control control;
+        public Transform inside;
+        public Transform outside;
+
+        void Start()
+        {
+            VRTK_InteractableObject io = GetComponent<VRTK_InteractableObject>();
+            if (io == null)
+            {
+                // treat as parent and assign to all children
+                foreach (VRTK_InteractableObject childIo in GetComponentsInChildren<VRTK_InteractableObject>())
+                {
+                    if (childIo.GetComponent<VRTK_ContentHandler>() == null)
+                    {
+                        VRTK_ContentHandler ch = childIo.gameObject.AddComponent<VRTK_ContentHandler>();
+                        ch.control = control;
+                        ch.inside = inside;
+                        ch.outside = outside;
+                    }
+                }
+            }
+        }
+
+        public void OnCollisionEnter(Collision collision)
```

Comment:
ah, no it's not private, it's internal visibility which means public inside the project. I tried to find out the guideline from Unity but was not successful. Definitely all MonoBehaviour methods are public (you can simply press F12 on MonoBehaviour), the Messages (Start, Awake) are not covered there though, they are magic. For sure they need to be more than private to work but what the exact visibility needs to be is a mystery. Unity creates Start/Awake with internal visibility. VS creates the remaining Messages (OnCollision...) with public visibility when selecting "Implement MonoBevahviours". I don't know enough scenarios to judge if it is safe to reduce the visibility of these from public to internal. Will that break scenarios where one could have reused a C# component from another project somehow? It's quite theoretical but still important to think about. What do you think? For now we should probably be fine by making them identical to Start/Awake and remove the visibility declaration.